searches/fibonacci_search.py

```diff
@@ -35,7 +35,17 @@ def fibonacci(k: int) -> int:
     5
     >>> print(fibonacci(15))
     610
+    >>> print(fibonacci('a'))
+    Traceback (most recent call last):
+    ValueError: k must be an integer.
+    >>> print(fibonacci(-5))
+    Traceback (most recent call last):
+    ValueError: k integer must be greater or equal to zero.
     """
+    if not isinstance(k, int):
+        raise ValueError("k must be an integer.")
+    if k < 0:
+        raise ValueError("k integer must be greater or equal to zero.")
```

Comment:
I have a question in this one.

Do you think is it better to use Assertions or this conditionals?

Reply:
Assertions will raise `AssertionError` which is ambiguous, the user has no idea what the function is asserting. By raising specific errors, users can understand and the code is easier to read.

You can use `try ... except ...` block but that would become cumbersome and unnecessary for such small cases.